src/commands.ts

```diff
@@ -0,0 +1,386 @@
+        // Add the connection reset option for webrtc
+        fields.splice(0,0, { prompt: "Reset connection" })
+    shell.t.writeln(`\x1B[4m${gate.name}\x1B[0m`)
+    let choice
+    try {
+        choice = await shell.newForm(fields, "menu")
+    } catch (e) {
+        return
+    }
+    let ans
+    switch (choice) {
+        case "Reset connection":
+            gate.disengage().then(async () => {
+                if (gate.session) {
+                    gate.session.close()
+                    gate.session = null
+                }
+                gate.t7.run(() =>  {
+                    gate.connect()
+                }, 100)
+            }).catch(() => gate.connect())
+            break
+        case "Reset connection & Layout":
+            try {
+                if (gate.session) {
+                    await gate.disengage()
+                    gate.session.close()
+                    gate.session = null
+                }
+                gate.connect(() => {
+                    gate.clear()
+                    gate.map.showLog(false)
+                    gate.activeW = gate.addWindow("", true)
+                    gate.focus()
+                })
+            } catch(e) {
+                gate.notify("Connect failed")
+            }
+            break
+        case "\x1B[31mFactory reset\x1B[0m":
+            try {
+                ans = (await shell.newForm(factoryResetVerify, "text"))[0]
+            } catch (e) {
+                return
+            }
+            if (ans == "y") {
+                gate.t7.factoryReset()
+                gate.clear()
+                gate.t7.goHome()
+            }
+            else
+                shell.map.showLog(false)
+            break
+        case "Close gate":
+            gate.boarding = false
+            gate.clear()
+            gate.updateNameE()
+            if (gate.session) {
+                gate.session.close()
+                gate.session = null
+            }
+            // we need the timeout as cell.focus is changing the href when dcs are closing
+            setTimeout(() => gate.t7.goHome(), 100)
+            break
+    }
+}
+
+async function editCMD (shell:Shell, args: string[]) {
+    const hostname = args[0]
+    if (!hostname)
+        return shell.t.writeln("Missing hostname")
+    const gates = terminal7.gates
+    const gate = gates.get(hostname)
+    if (!gate)
+        return shell.t.writeln(`Host not found: ${hostname}`)
+    const f1 = [
+        { prompt: "Connect" },
+        { prompt: "Edit" },
+        { prompt: "\x1B[31mDelete\x1B[0m" },
+    ]
+    let f2 = [
+        {
+            prompt: "Name",
+            default: gate.name,
+            validator: a => gate.t7.validateHostName(a)
+        },
+        { 
+            prompt: "Hostname",
+            default: gate.addr,
+            validator: a => gate.t7.validateHostAddress(a)
+        },
+        { prompt: "Username", default: gate.username }
+    ]
+    const fDel = [{
```

Comment:
There's inconsistency in form names. With use `f1`, `f2`, `f3` or better, `fMain`, `fFields`, `fDel`